File: CS50/sentimental-mario-more/mario.py
# TODO
# Implement a program that prints out a double half-pyramid of a specified height

from cs50 import get_int

# Prompt user for input with loop
while True:
    height = get_int("Heigth: ")

    # Check if input is valid with conditions; loop again if not valid
    if not height > 0:
        continue
    if not height < 9:
        continue

    # Else exit the loop
    else:
        break

# Start main loop for printing
for x in range(1, height + 1):

    # Use print(... end="") to stay in same line. Start with padding left...
    print(" " * (height - x), end="")

    # Place bricks (hashes #) left
    print("#" * x, end="")

    # Middle space of 2
    print("  ", end="")

    # Place bricks (hashes #) right
    print("#" * x)

File: CS50/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height, hashes, row, spaces;

    // Prompt user for input
    do
    {
        height = get_int("Set heigth: \n");
    }
    while (height < 1 || height > 8);

    // Main loop
    for (row = 0; row < height; row++)
    {
        // Draw left spaces
        for (spaces = row + 1; spaces < height; spaces++)
        {
            printf(" ");
        }

        // Draw left hashes
        for (hashes = 0; hashes < row + 1; hashes++)
        {
            printf("#");
        }

        // Draw divider
        printf("  ");

        // Draw right hashes & end line break
        for (hashes = 0; hashes < row + 1; hashes++)
        {
            if (hashes == row)
            {
                printf("#\n");
                break;
            }
            printf("#");
        }
    }
}